constraint/pkg/instrumentation/instrumentation.go

```diff
@@ -0,0 +1,84 @@
+// package instrumentation defines primitives to
+// support more observability throughout the contraint framework.
+package instrumentation
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+)
+
+// Stat is a Name, Value, Description tuple that may contain
+// metrics or aggregations of metrics.
+type Stat struct {
+	Name        string `json:"name"`
+	Value       Value  `json:"value"`
+	Description string `json:"description"`
+}
+
+// StatsEntry comprises of a generalized Key for all associated Stats.
+type StatsEntry struct {
+	Key   interface{} `json:"key"`
+	Stats []*Stat     `json:"stats"`
+}
+
+type Value interface {
+	String() string
```

Comment:
IIRC all golang types support String() one way or another (I may be wrong about this), but generally are string-ifiable. What is the benefit of Value() over just `interface{}`?

Reply:
I think the main selling point for me, to my mind, was to trim the float64 up to 2 decimal points. I can also buy the argument to leave that up to the consumer after a type cast.

Reply:
I thought about it some more and I think we can leave it up to consumers to decide how to typecast; I think it's cleaner from the CF end.